### PR TITLE
Simplify logic in ScoreCachingWrappingScorer

### PR DESCRIPTION
**Pull request by msfroh**
### Description

This is functionally equivalent to the logic that was present, but makes the behavior clearer.

### CODE REVIEW

File: lucene/core/src/java/org/apache/lucene/search/ScoreCachingWrappingScorer.java [hunk 1]
```diff
@@ -31,8 +31,7 @@
  */
 public final class ScoreCachingWrappingScorer extends Scorable {
 
-  private int lastDoc = -1;
-  private int curDoc = -1;
+  private boolean computeScore;
```
Comment (vigyasharma) [inline lucene/core/src/java/org/apache/lucene/search/ScoreCachingWrappingScorer.java:34]: Minor: how about renaming this to something that tells whether the score is cached already e.g. `scoreCached` ?And the flag gets unset whenever we collect a new doc (we would reverse the boolean meaning with this name).

**Review by vigyasharma — APPROVED** [review]
I like this change. It's the first time I read this class, and this PR did make it easier for me. Any reason this is still in draft?

File: lucene/core/src/java/org/apache/lucene/search/ScoreCachingWrappingScorer.java (hunk 1 above)
Comment (jpountz) [reply lucene/core/src/java/org/apache/lucene/search/ScoreCachingWrappingScorer.java:34]: +1 I had a similar reaction when reading this PR. It's easier for me to reason about when the cached score should be invalidated than when the score should be computed.

**Review by jpountz — APPROVED** [review]
Nice simplification.

**Comment by msfroh** [comment]
Thanks folks! I made the suggested change and took this PR out of draft. 

I think this one makes more sense than my previous one over at https://github.com/apache/lucene/pull/13987

**Review by vigyasharma — APPROVED** [review]
Looks good, @msfroh.. Can you also add a Changes entry?

**Review by jpountz — APPROVED** [review]

**Comment by jpountz** [comment]
@msfroh Do you think you'll have time to add a CHANGES entry? Otherwise I'll do it for you and merge.

**Comment by msfroh** [comment]
> Looks good, @msfroh.. Can you also add a Changes entry?

I wasn't planning to add a CHANGES entry. Nothing functionally changes, there's no optimization. It's just an extremely minor code cleanup.

**Comment by jpountz** [comment]
Fair.

**Pull request merged**